src/proj_equirect.jl

```diff
@@ -327,31 +316,48 @@ function mapblocks(fun::Function, Ms::BlockDiagEquiRect{B}...) where {B<:Basis}
     map(M->promote_metadata_strict(M.proj, Ms[1].proj), Ms) 
     BlockDiagEquiRect{B}(
         map(
-            i->fun(getindex.(getproperty.(Ms,:blocks),:,:,i)...), 
+            i->fun(getindex.(getproperty.(Ms,:blocks),:,:,i)...), # This looks miserable:(
             axes(Ms[1].blocks,3),
         ),
         Ms[1].proj,
     )
 end 
 
-# TODO: add simulate
-
 # ## Other methods 
 
 function adapt_structure(storage, L::BlockDiagEquiRect{B}) where {B}
     BlockDiagEquiRect{B}(adapt(storage, L.blocks), adapt(storage, L.proj))
 end
-# function adapt_structure(storage, L::BlockDiagEquiRect{B}) where {B}
-#     BlockDiagEquiRect{B}(adapt(storage, L.blocks), adapt(storage, L.blocks_sqrt), adapt(storage, L.proj))
-# end
 
 # TODO: summary methods for BlockDiagEquiRect{B} and Adjoint{T,BlockDiagEquiRect{B}}
 
+# Turning this on to see if it helps testing with ≈
+size(L::BlockDiagEquiRect) = (fill(L.proj.Nx * L.proj.Ny, 2)...,)
+
+function simulate(rng::AbstractRNG, M::BlockDiagEquiRect{AzFourier,ProjEquiRect{T}}) where {T}
+    @unpack Ny, Nx = M.proj
+    spin0_whitepix_mat = randn(T, M.proj.Ny, M.proj.Nx) # Note: T should be real here. Do we add T<:Real to method sig? 
+    spin0_whitepix_fld = EquiRectMap(spin0_whitepix_mat, M.proj)
+    mapblocks(M, spin0_whitepix_fld) do Mb, vb 
+        sqrt(Hermitian(Mb)) * vb
+    end
+end
 
+function simulate(rng::AbstractRNG, M::BlockDiagEquiRect{QUAzFourier,ProjEquiRect{T}}) where {T}
+    @unpack Ny, Nx = M.proj
+    spin2_whitepix_mat = randn(Complex{T}, M.proj.Ny, M.proj.Nx)
+    spin2_whitepix_fld = EquiRectQUMap(spin2_whitepix_mat, M.proj)
+    mapblocks(M, spin2_whitepix_fld) do Mb, vb 
+        sqrt(Hermitian(Mb)) * vb
+    end
+end
 
+#TODO: it would be nice to have a simulate method that produced white noise ... 
```

Comment:
Agree, here's the implementation for Lambert:

https://github.com/marius311/CMBLensing.jl/blob/b5b8b3ebdef8f86cc749d88ca72c544aea7a2c6f/src/proj_lambert.jl#L347-L352